src/python/Services/ReqMgr/ReqMgrReader.py

```diff
@@ -6,53 +6,383 @@
 
 import logging
 import os
+import time
+import copy
 from Utils.WebTools import getResponse
 from Utils.ConfigurationHandler import ConfigurationHandler
 
+from typing import List, Optional, Union, Any, Callable
+
 
 class ReqMgrReader(object):
     """
     _ReqMgrReader_
     General API for reading data from ReqMgr
     """
 
-    def __init__(self, url=None, logger=None, **contact):
-
+    def __init__(
+        self, url: str = None, logger: Optional[logging.Logger] = None, **contact
+    ) -> None:
+        # TODO: url is not being used
         try:
             configurationHandler = ConfigurationHandler()
-            self.reqmgrUrl = os.getenv('REQMGR_URL', configurationHandler.get('reqmgr_url'))
+            self.reqmgrUrl = os.getenv(
+                "REQMGR_URL", configurationHandler.get("reqmgr_url")
+            )
+            self.reqmgrEndpoint = {
+                "req": "/reqmgr2/data/request/",
+                "info": "/reqmgr2/data/info/",
+                "agentConfig": "/reqmgr2/data/wmagentconfig/",
+                "splitting": "/reqmgr2/data/splitting/",
+                "couchdb": "/couchdb/reqmgr_workload_cache/",
+            }
+
             logging.basicConfig(level=logging.INFO)
             self.logger = logger or logging.getLogger(self.__class__.__name__)
-        except Exception as e:
+
+        except Exception as error:
             msg = "Error initializing ReqMgrReader\n"
-            msg += "%s\n" % format(e)
+            msg += f"{error}\n"
             raise Exception(msg)
 
-    def getWorkflowByCampaign(self, campaign, details=False):
+    def _runWithRetries(
+        self,
+        fcn: Callable,
+        fcn_pargs: list,
+        fcn_args: dict = {},
+        default: Optional[Any] = None,
+        tries: int = 10,
+        wait: int = 5,
+    ) -> Any:
+        # TODO: is there a better place to put this function?
```

Comment:
How about in a new module called `RetryManager.py` under `src/python/Utils/`? It's for sure that it should not be in ReqMgrReader.

Reply:
My experience is that the system became much more robust since it was initially implemented. So I'd be inclined to drop these HTTP retry call logic, it will make the code more clear and maintainable.

Reply:
yes! Totally agree: https://github.com/CMSCompOps/WmAgentScripts/pull/859/files#r658569903